sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/MergeScalarSubqueries.scala

```diff
@@ -21,7 +21,7 @@ import scala.collection.mutable
 import scala.collection.mutable.ArrayBuffer
 
 import org.apache.spark.sql.catalyst.expressions._
-import org.apache.spark.sql.catalyst.expressions.aggregate.AggregateExpression
+import org.apache.spark.sql.catalyst.expressions.aggregate.{AggregateExpression}
```

Comment:
Why change this line ?

Reply:
I don't know what to do with this PR. There doesn't seem to be much interrest in this improvement from the community, but I'm happy to fix this if we can move forward somehow...

My plan was:
1.  to allow filter merging for subqueries in this PR,
2. and then extract the merging logic to be able to apply it on other areas of the plan,
3. and then apply it on other areas like https://github.com/apache/spark/pull/40661.

Reply:
I like this optimization, and it has already been migrated into our company's internal branch by me. User cases similar to the tpcds q9 scenario will stand to benefit.

Reply:
Yes. I merged this PR into our private repository half years ago. I also want to promote this PR.

Reply:
Fixed in https://github.com/apache/spark/pull/37630/commits/563cef99dccd8d7afea5bc09a2caaadfdde27aed.

Reply:
I've updated the PR with the latest `master` and checks passed.
Anyone would like to review/approve this PR?